internal/internal_task_pollers.go

```diff
@@ -164,6 +176,42 @@ func (lat *localActivityTunnel) sendTask(task *localActivityTask) bool {
 	}
 }
 
+func newLocallyDispatchedActivityTunnel(stopCh <-chan struct{}) *locallyDispatchedActivityTunnel {
+	return &locallyDispatchedActivityTunnel{
+		taskCh: make(chan *locallyDispatchedActivityTask, 25),
```

Comment:
For now they don't share the same worker/processor, so there should be no contention. But this could happen if we try to combine the pollers (see comment above).
The severity of this issue is something we need to verify. There are various ways to mitigate this if it happens (e.g. add heuristics to decide whether to poll remotely or not, add cancellation logic after poll, etc).

Reply:
Yes making this configurable is a good suggestion.